2.4/Dockerfile

```diff
@@ -38,7 +34,14 @@ RUN yum install -y epel-release && \
     yum clean all && \
     mkdir -p /var/lib/mongodb/data && chown -R mongodb.mongodb /var/lib/mongodb/ && \
     test "$(id mongodb)" = "uid=184(mongodb) gid=998(mongodb) groups=998(mongodb)" && \
-    chmod o+w -R /var/lib/mongodb && chmod o+w -R /opt/rh/mongodb24/root/var/lib/mongodb
+    # Loosen permission bits to avoid problems running container with arbitrary UID
+    chmod -R a+rwx /opt/rh/mongodb24/root/var/lib/mongodb
+
+COPY run-mongod.sh /usr/local/bin/
+COPY contrib /var/lib/mongodb/
+
+# Loosen permission bits to avoid problems running container with arbitrary UID
+RUN chmod -R a+rwx /var/lib/mongodb
```

Comment:
yeah i'm not sure the best approach here either.  i don't love making non executable files executable, but it'd be worse to leave a file that should be executable, not able to be executed (though in theory a user could make a copy of it and make the copy executable).

i think for now a+rwx is the safer course of action, but long term we need a better solution for permissions in general.